### ISSUE TITLE
Color Formatting Hex/Hash Assignment (enhancement)

### ISSUE THREAD

**Issue by MarcGJA** [metabase/metabase#9682]
Allow users to choose whatever colors they wish for visualizations elements & bins in charts and maps by giving the option to use <b>hex/hash</b> color assignments rather than preset color palettes.  Maps in particular are rather mono-tone having shades that are difficult for many users to distinguish readily and are especially problematic for those with vision impairments.  In addition to being able to customize binning (<a href="https://github.com/metabase/metabase/issues/9208">#9208</a>) multi-colored maps would be a very helpful enhancement.

The idea has come up before in <a href="https://github.com/metabase/metabase/pull/2737">GitHub Pull #2737 pt 6</a> but it doesn't appear anything ever came of it.  And something similar tied to color-pin maps (<a href="https://discourse.metabase.com/t/change-the-color-of-pins-location-pins-in-world-map/5156">Forum #5156</a>).  But there doesn't appear to be a GitHub issue on this topic until now.

In short, shift the color formatting burden off of Metabase and onto users.  That way Metabase doesn't have to design, maintain, or spend valuable time and resources on dealing with color-palettes at all.  And users can have unlimited choices on how they want to color-format visualizations.

It's a win-win!

**Comment by MarcGJA** [comment]
Note that <b>hex/hash</b> color assignments could even be associated to a secondary field say for profit (1) / loss (1) as mentioned in <a href="https://discourse.metabase.com/t/change-the-color-of-pins-location-pins-in-world-map/5156">Forum #5156</a>with regard to color-coding a pin-map.

**Comment by MarcGJA** [comment]
Note that <b>hex/hash</b> color assignments could even be associated to a secondary field say for profit (1) / loss (1) as mentioned in <a href="https://discourse.metabase.com/t/change-the-color-of-pins-location-pins-in-world-map/5156">Forum #5156</a> with regard to color-coding a pin-map.

**Comment by flamber** [comment]
Given that colors are needed for x-rays and drill-down, then Metabase would need to maintain a palette of nice colors to automatically assign.
But I would really love to be able to manually add other colors - or just the option to do custom colors per chart.
Maybe something like what most paint tools do, where it allows you to select default colors, while still allowing you to set a custom color for a specific thing.
This is Gimp - I know the interface is a bit over the top, but just as an example:
![image](https://user-images.githubusercontent.com/1447303/55507006-52e5c480-5646-11e9-89f8-86513cb9cf70.png)

**Comment by MarcGJA** [comment]
Thanks @flamber, and it doesn't have to be elaborate, just a fill-in box to associate the actual hex-code to an element or bin-category would be enough.  There are a multitude of exceedingly helpful websites out there like https://coolors.co/ and https://www.color-hex.com/ dedicated just to choosing a hex/hash based color-pallet.  It would definitely be worth it for all Metabase users and would make Metabase stand out among BI tools!

**Comment by MarcGJA** [comment]
<b>Hex/hash</b> color assignments could even be associated to a secondary field say for profit (1) / loss (0) as mentioned in <a href="https://discourse.metabase.com/t/change-the-color-of-pins-location-pins-in-world-map/5156">Forum #5156</a> with regard to color-coding a pin-map.

**Comment by salsakran** [comment]
Just a heads up - `shift the color formatting burden off of Metabase and onto users` is pretty much the exact opposite of what we're trying to do with Metabase.

We're into adding knobs as needed, but our entire premise is _not_ having to specify every last thing and to JustWork out of the box

**Comment by maccca** [comment]
A useful alternative could be to allow for a custom color palette to be provided in the Admin section of Metabase?

The default colors in this palette would be the existing color set, but would allow for a customisation of the palette. I would find this extremely useful as it compliments embedding nicely with existing applications.

**Comment by DataDrivenAngel** [comment]
Being able to expand the color palette would be great. 

I frequently find myself wishing for the ability to have several different shades of a color to indicate that clusters of points or lines are related.

**Comment by joshpk** [comment]
Are there plans to expand the availability of colours? For us its a major limitation not being able to customize colours on graphs.

**Comment by flamber** [comment]
The Enterprise Edition has options to change the color palette: https://www.metabase.com/learn/developing-applications/advanced-metabase/brand.html

**Comment by KayJey-Krayen** [comment]
Any update on adding extended color palette on metabase?
